src/Core/Checkout/Cart/CartRuleLoader.php

```diff
@@ -81,6 +81,10 @@ public function invalidate(): void
     private function load(SalesChannelContext $context, Cart $cart, CartBehavior $behaviorContext, bool $new): RuleLoaderResult
     {
         return Profiler::trace('cart-rule-loader', function () use ($context, $cart, $behaviorContext, $new) {
+            // If the processing starts with deferred errors already in the cart, the cart MUST be persisted
+            // to remove the errors from the stored cart
+            $hasDeferredErrors = $cart->getErrors()->count() > 0;
```

Comment:
This should also be put behind/include the feature flag, as this could change existing behaviour